src/test/java/com/example/rets_api/repository/BasicPropertyWithOneEntityRepositoryJPATest.java

```diff
@@ -0,0 +1,183 @@
+
+        //Validating
+        checkAllBasicFieldsFromProperty(propertySaved);
+
+        assertNotNull(propertySaved.getFinancialData());
+        assertTrue(CompareEntitiesTest.compareFinancialData(propertySaved.getFinancialData(), propertyEntityToCompare.getFinancialData()));
+        assertNotNull(propertySaved.getFinancialData().getFinancialDataId());
+    }
+
+    @Test
+    public void should_store_a_PropertyAndRoom() {
+        //Creating
+        PropertyEntity propertyToSave = EntityUtilsTest.createPropertyEntityWithBasicFieldsAndRoom();
+        PropertyEntity propertyEntityToCompare = EntityUtilsTest.createPropertyEntityWithBasicFieldsAndRoom();
+
+        //Saving
+        PropertyEntity propertySaved = propertyRepository.saveAndFlush(propertyToSave);
+
+        //Validating
+        checkAllBasicFieldsFromProperty(propertySaved);
+
+        assertNotNull(propertySaved.getRoomList());
+        assertEquals(propertySaved.getRoomList().size(), propertyEntityToCompare.getRoomList().size());
+        assertTrue(CompareEntitiesTest.compareRoom(propertySaved.getRoomList().get(0), propertyEntityToCompare.getRoomList().get(0)));
+        assertTrue(CompareEntitiesTest.compareRoom(propertySaved.getRoomList().get(1), propertyEntityToCompare.getRoomList().get(1)));
+        assertNotNull(propertySaved.getRoomList().get(0).getRoomId());
+        assertNotNull(propertySaved.getRoomList().get(1).getRoomId());
+    }
+
+    @Test
+    public void should_store_a_PropertyAndSchool() {
+        //Creating
+        PropertyEntity propertyToSave = EntityUtilsTest.createPropertyEntityWithBasicFieldsAndSchool();
+        PropertyEntity propertyEntityToCompare = EntityUtilsTest.createPropertyEntityWithBasicFieldsAndSchool();
+
+        //Saving
+        PropertyEntity propertySaved = propertyRepository.saveAndFlush(propertyToSave);
+
+        //Validating
+        checkAllBasicFieldsFromProperty(propertySaved);
+
+        assertNotNull(propertySaved.getSchoolList());
+        assertEquals(propertySaved.getSchoolList().size(), propertyEntityToCompare.getSchoolList().size());
+        assertTrue(CompareEntitiesTest.compareSchool(propertySaved.getSchoolList().get(0), propertyEntityToCompare.getSchoolList().get(0)));
+        assertNotNull(propertySaved.getSchoolList().get(0).getSchoolId());
+    }
+
+    @Test
+    public void should_store_a_PropertyAndCommunity() {
+        //Creating
+        PropertyEntity propertyToSave = EntityUtilsTest.createPropertyEntityWithBasicFieldsAndCommunity();
+        PropertyEntity propertyEntityToCompare = EntityUtilsTest.createPropertyEntityWithBasicFieldsAndCommunity();
+
+        //Saving
+        PropertyEntity propertySaved = propertyRepository.saveAndFlush(propertyToSave);
+
+        //Validating
+        checkAllBasicFieldsFromProperty(propertySaved);
+
+        assertNotNull(propertySaved.getCommunity());
+        assertEquals(propertySaved.getCommunity(), propertyToSave.getCommunity());
```

Comment:
assertEquals(propertySaved.getCommunity(), propertyToSave.getCommunity()); is unecessary because we are using assertTrue(CompareEntitiesTest.compareCommunity(propertySaved.getCommunity(), propertyEntityToCompare.getCommunity()));